homeassistant/components/caldav/calendar.py

```diff
@@ -103,16 +107,14 @@ def setup_platform(
             name = cust_calendar[CONF_NAME]
             device_id = f"{cust_calendar[CONF_CALENDAR]} {cust_calendar[CONF_NAME]}"
             entity_id = generate_entity_id(ENTITY_ID_FORMAT, device_id, hass=hass)
-            calendar_devices.append(
-                WebDavCalendarEntity(
-                    name=name,
-                    calendar=calendar,
-                    entity_id=entity_id,
-                    days=days,
-                    all_day=True,
-                    search=cust_calendar[CONF_SEARCH],
-                )
+            coordinator = CalDavUpdateCoordinator(
```

Comment:
If there's one coordinator per entity, why should we use a coordinator?

Reply:
I agree it is not strictly necessary given the results are not shared between entities. I'll give some thoughts below and happy to adjust the plan based on your feedback.

My thinking is that `WebDavCalendarData` is effectively an update coordinator, so it would be an improvement to use a known abstraction.

The primary motivation originally for that data class was probably copied from google calendar where the goal was to have one interval for fetching data from the server (the `@Throttle(MIN_TIME_BETWEEN_UPDATES)` below)  and a separate interval for noticing state transitions (the entity has its own default 1 minute update interval). That pattern is no longer needed after #98037 so additionally by moving to `CoordinatorEntity` this is fixing that.

Some upcoming considerations:
- I'd like to move this to the UI to be able to support additional platforms like todo. (Not sure that this coordinator will be reused for that, however)
- We probably won't want to support to many `custom_calendars` options as integration specific options anymore. Therefore, I was thinking we'd want some way to limit the scope of that similar to google calendar, so the data fetching is separate from the entity. We *might* want separate entity classes that drop some of that extra complexity, offsets, etc.

Some alternatives are:
- Move everything into the entity
- Move to some other client library or utility class that is a `caldav.DAVClient` wrapper that helps with the filtering and data type converstion
- Leave it alone and deal with the complexity of these cases together without separation

Let me know your thoughts.

Reply:
It's ok to use the coordinator. We just don't normally do that when there's one entity per coordinator.